src/core/datastore/syncstore.spec.js

```diff
@@ -128,6 +128,51 @@ describe('SyncStore', () => {
         });
     });
 
+    it('should throw an error if the tag is not a string', () => {
+      expect(() => {
+        new SyncStore(collection, { tag: {} });
+      }).toThrow();
+    });
+
+    it('should throw an error if the tag is an emptry string', () => {
+      expect(() => {
+        new SyncStore(collection, { tag: ' ' });
+      }).toThrow();
+    });
+
+    it('should return the entities by tag', () => {
```

Comment:
If we're naming this a `spec` file, we should define the full spec in tests. I (as you know) am not a huge fan of unit tests, but we're either doing it, or not. And by this I mean the behaviour for sync-related tasks should be in the spec as well.

Reply:
Are you referring to the spec for this PR?

Reply:
@tejasranade What are your thoughts on this?

Reply:
If we're talking about the tests to cover this feature, I do think we need some more tests here. 

My test case was intended to validate our approach. To release this, we should test beyond the base case. In the very least, we should validate a couple of additional typical scenarios - 
- saving data to one of the tagged datastores, followed by a sync.
- clearing data from one of the datastores, and making sure the other isn't cleared 

Both of the above are actually customer expectations for this feature, so I fully expect those to get used.

Reply:
As a side note - my `foo` / `bar` naming scheme shouldn't make it to master 😄 . We use `randomString()` in all our tests, and let's keep it that way!

Reply:
I mean in general. Calling these `.spec` files is a TDD convention, no? If that's what we're doing, we should be doing it. It's not just renaming a file. If we're not changing the way we treat those, we might as well leave them as `.test` files.

Reply:
This is out of scope for this PR and should be done in another PR.

Reply:
I added unit tests for saving date to one of the tagged datastore, followed by a push and validating another datastore did not get pushed.

I added a unit tests to clear one tagged datastore and validated that another one did not get cleared.

Reply:
I'm not really pushing for a full suite of tests, as much as I'd like us to start developing a specification for a feature before we start working on it. I feel this is a much needed improvement.

Reply:
@georgiwe if you're referring to TDD, we don't do that as a process today. The spec is supposed to be part of the JIRA ticket description.
Re: what to name the test file, that's a matter of convention and not process, as far as I can see. It's worth following a common convention, but that should be a separate breakout discussion (log it in JIRA? 😄 )